### ISSUE TITLE
Coin and document effective ACL URI

### ISSUE THREAD

**Issue by bblfish** [solid/specification#325]
There are some very good reasons to have default ACLs (WAC). There are also very good reason for each resource to have its own ACL (ACP). How can these two points of views come together?
Simply by allowing LDPRs to distinguish between two ACL resources

 1) A link to the effective Access Control Resource
 2) A link to the resource's potential Access Control Resource

Indeed the current [WAC spec](https://solid.github.io/web-access-control-spec/) states that

> a separate link relation type targeting the effective ACL resource is allowed, but no behaviour is defined by this specification.

Please specify this officially!

[Trellis has defined](https://github.com/solid/specification/issues/106#issuecomment-631410831) `http://www.trellisldp.org/ns/trellis#effectiveAcl`. It should be possible to have one that we all agree on in a solid namespace.

## Problems solved

The problems are:

* If only 1) is given then there is no way to create subspaces on a server with different access control rules, as there is no way to work out where the resource's acl will be
* if only 2) is given but the default acl needs to be discovered (as per current WAC) then the [cost of finding the effective ACL is 2n+1](https://github.com/solid/web-access-control-spec/issues/99) where n is the number of folders between the resource and the effective one. This is unworkable.
* If every resource has 2 be the effective resource - as ACP does - then we have a problem of wastage and maintenance. This creates duplication of information as per [issue 206 of the authorization panel](https://github.com/solid/authorization-panel/issues/206). For all those cases where a default set of rules would do for most content, one ends up with a duplication of rules. Every resource will end up with the same copy of pretty much the same ACL. This creates problems for updating them, and also would slow down requests, fill up intermediate caches with duplicate information, etc... So it can be pretty wasteful. It slows down requests that are access controlled, by requiring essentially 1 more request than needed. 

The answer to both these problems, is just to coin a new Link relation as discussed in last weeks authorization panel meeting
https://github.com/solid/authorization-panel/blob/draft-minutes/meetings/2021-10-13.md

**Comment by csarven** [comment]
Can we move any new information in this issue to https://github.com/solid/web-access-control-spec/issues/99 and then close this issue? or is the request in this repo to have one that can be used by any access control system?

**Comment by bblfish** [comment]
Well there is a default currently and that is to use `http://www.trellisldp.org/ns/trellis#effectiveAcl`. The WAC spec should mention that, since its aim is to describe existing implementations.

```turtle
@prefix trellis: <http://www.trellisldp.org/ns/trellis#> .
trellis:effectiveAcl     a rdf:Property;
         :comment "The ACL that currently controls the resource in question."@en;
         :isDefinedBy t:;
         :label "effective ACL"@en;
         vs:term_status "working-draft" .
```

**Comment by RubenVerborgh** [comment]
> Do that study deal with clients needing to do access control?

No, with an even more complex Linked Data case that involves much more than 2n+1 and we can pull it off just fine.
So this shows that theoretical evidence is insufficient, and that we need empirical evidence that this simpler problem would pose a performance bottleneck in practice.

> In any case having a link that will in many situations always fail

You have a different definition of "fail".

The fact that it is a 404 is seen as a feature in Solid; a signal that there is no resource-specific ACL and that the agent is able to create one at that place if they so desire.

I will halt the discussion here; I think I've sufficiently made my point: we need server implementers in favor, and my vote can be bought with real-world measurements.

**Comment by bblfish** [comment]
If we can't come up with a good URL for the effective access control resource, I suggest we use `wac:accessControl`, i.e. `http://www.w3.org/ns/auth/acl#accessControl` to link to the effective access control resource. That would require a little change in the definition of that relation.
